client/my-sites/post-selector/selector.jsx

```diff
@@ -95,6 +101,11 @@ const PostSelectorPosts = React.createClass( {
 				requestedPages: [ 1 ]
 			} );
 		}
+
+		if ( this.props.posts !== nextProps.posts ) {
```

Comment:
I see we are comparing posts here and in line 121 - is performing this comparison a bit intensive due to posts being an array of large objects?